image/helpers.go

```diff
@@ -209,8 +209,10 @@ func NewToolingStore() (*ToolingStore, error) {
 
 // DownloadOptions carries options for downloading snaps plus assertions.
 type DownloadOptions struct {
+	TargetDir  string
+	TargetFunc func(*snap.Info) (string, error)
```

Comment:
Could you please document what are the semantics of TargetDir and TargetFunc and how do they play together (it's not immediately obvious looking at the `String()`).

Reply:
Without having read further I wonder if TargetFunc() refers to a target filename or a target dir, maybe this could be made a bit more obvious?